pkg/_pub_shared/lib/data/task_payload.dart

```diff
@@ -70,4 +71,6 @@ class VersionTokenPair {
   factory VersionTokenPair.fromJson(Map<String, dynamic> json) =>
       _$VersionTokenPairFromJson(json);
   Map<String, dynamic> toJson() => _$VersionTokenPairToJson(this);
+
+  late final semanticVersion = Version.parse(version);
```

Comment:
nit.
```suggestion
  /// [version] parsed as [Version] object.
  late final semanticVersion = Version.parse(version);
```

Honestly, I'm tempted to do:
```suggestion
  /// [version] as [Version] object.
  Version get semanticVersion => Version.parse(version);
```